### DISCUSSION TITLE
Move and restore focus without containing it

### DISCUSSION THREAD

**Discussion by filipsvensson** [adobe/react-spectrum#7198] [Q&A]
Hi!
I'm trying to use the FocusScope component to move and restore focus but without containing the focus `<FocusScope contain={false} restoreFocus autoFocus>` 
Looking at [useRestoreFocus](https://github.com/adobe/react-spectrum/blob/987f174ba5a0a7d65a7e07acec5cc832aae4abdb/packages/%40react-aria/focus/src/FocusScope.tsx#L571) it looks like it's only active if contain is true. Is this something that could be changed?

**Comment by snowystinger** [comment]
Can you share something demonstrating what you're trying to build? Maybe we'll have other suggestions.

**Reply by filipsvensson** [reply]
![image](https://github.com/user-attachments/assets/5e7b60ce-0e4d-4fd9-8981-d6b7c9f17df2)
I have a 3D scene with focusable 3D modules. Selecting a 3D model will open a menu, but there is also a "root menu" open when no 3D model i selected. This menu has options that are global and not related to a specific model. 
Both menus can open more menus like the info menu shown in the image.
I want to move the focus between these menus and between the 3D scene and the menus without locking the focus inside them. And then restore the focus if the user closes a menu.

**Reply by snowystinger** [reply]
Thanks for sharing the extra information. You'll want to manage focus restoring yourself. Our restore focus is currently tied to contain as you've noted. Part of this is that we only mark where the focus came from once upon entering. This eliminated a class of bugs around the restore target components remounting/unmounting/etc while the user is inside the focusscope.

When the menu closes, restore focus to wherever makes the most sense for your user.

**Reply by filipsvensson** [reply]
Ok! Thanks for your answer
